packages/clerk-js/src/ui/contexts/ClerkUIComponentsContext.tsx

```diff
@@ -111,26 +114,22 @@ export const useSignInContext = (): SignInContextType => {
   }
 
   const afterSignUpUrl = clerk.buildUrlWithAuth(
-    pickRedirectionProp('afterSignUpUrl', {
-      queryParams,
-      ctx,
-      options,
-      displayConfig,
-    }),
+    pickUrl(['afterSignUpUrl', 'redirectUrl'], queryParams, {
+      validator: url => isAllowedRedirectOrigin(url, options.allowedRedirectOrigins),
+      formatter: camelToSnake,
+    }) || pickUrl(['afterSignUpUrl', 'redirectUrl'], [ctx, options, displayConfig]),
   );
 
   const afterSignInUrl = clerk.buildUrlWithAuth(
-    pickRedirectionProp('afterSignInUrl', {
-      queryParams,
-      ctx,
-      options,
-      displayConfig,
-    }),
+    pickUrl(['afterSignInUrl', 'redirectUrl'], [queryParams], {
+      validator: url => isAllowedRedirectOrigin(url, options.allowedRedirectOrigins),
+      formatter: camelToSnake,
+    }) || pickUrl(['afterSignInUrl', 'redirectUrl'], [ctx, options, displayConfig]),
```

Comment:
☁️ i think that it's a great idea to refactor the `pickRedirectionProp` to something simpler and more maintainable than currently is but the above code (personal preference) does not make it easier for me to understand the rules of picking the `afterSignInUrl`. :(